### ISSUE TITLE
Compilation error due optional-lite non-standard library

### ISSUE THREAD

**Issue by patham9** [trueagi-io/hyperon-experimental#564]
There is a new breakage somewhere in the interplay between the many build systems cargo, conan, cmake, make, It was necessary to manually download optional-lite and to put the header file locally, which overall led to the following modifcations which also affects a cpp file:

![image](https://github.com/trueagi-io/hyperon-experimental/assets/8284677/84cc911f-b840-4dcc-963f-d24274897e92)

**Comment by patham9** [comment]
This particular issue only occured on MacOS Ventura, not on Ubuntu as I tried today.

**Comment by Necr0x0Der** [comment]
Hmm... I'm using MacOS Ventura on M1 and it is still working for me, and no breaking changes in this regard happened to me. But conan is capricious. It has rebuilt all the dependences for me recently.
What will happen if you remove ~/.conan and repeat
```bash
python3 -m pip install conan==1.60.2
conan profile new --detect default
```
?
And what version of conan are you using?

**Comment by vsbogd** [comment]
We have to use `optional-lite` because it allows install just small library instead of full `boost` installation and it works for the old versions of the C++ compilers which doesn't have `std::optional` in standard library. We need supporting old compilers because they are used to release the Python package for some platforms. If you have modern compiler you can even import optional from `std` I believe as a workaround.

I see `optional-lite` is still in ConanCenter repo which we use to get 3rdparty C/C++ artifacts (https://conan.io/center/recipes/optional-lite?version=3.5.0). So Conan should be able to download it. Could you please explain a case a bit. Do you try to run fresh installation or your previous installation is broken after update?

Generally what Alexey suggested in a [previous comment](https://github.com/trueagi-io/hyperon-experimental/issues/564#issuecomment-1922172165) should work. It clears the cache of the Conan and should download dependencies 
again. If it doesn't work, could you please attach a log of the `cmake` and `make` commands to the issue?

**Comment by patham9** [comment]
@Necr0x0Der: Now I see, it seems I had used Brew version 1.62.0 from Homebrew since
`python3 -m pip install conan==1.60.2`
did not leave my system with an executable conan executable, also not when the command is executed with sudo, I guess it is another case where pip messes up? The reliability of pip seems unfortunately lacking in various occasions, which is not your fault.
For completeness, the C++ compiler: Apple clang version 14.0.3 (clang-1403.0.22.14.1)

**Comment by vsbogd** [comment]
> since `python3 -m pip install conan==1.60.2` did not leave my system with an executable conan executable, also not when the command is executed with sudo, I guess it is another case where pip messes up? 

Hm, yes it is really strange. I tried it on a fresh Ubuntu: first I installed `conan==1.62` using pip then I switched back to the `1.60.2`. In both cases `conan` executable is present (Conan itself is a Python app). If you don't have `conan` executable then it may be a reason why build doesn't work.

> For completeness, the C++ compiler: Apple clang version 14.0.3 (clang-1403.0.22.14.1)

Compiler seems absolutely ok to me. Despite it supports `C++17` code is written to use `C++11` and that's why the `nostd/optional.hpp` library is used.
